src/features/defi/helpers/utils.ts

```diff
@@ -38,6 +44,19 @@ export const canCoverTxFees = ({
   return bnOrZero(feeAssetBalanceCryptoHuman).minus(bnOrZero(estimatedGasCryptoPrecision)).gte(0)
 }
 
+export const getFeeDataFromEstimate = ({
+  average,
+  fast,
+}: FeeDataEstimate<EvmChainId>): FeeData<EvmChainId> => ({
+  txFee: fast.txFee, // use worst case tx fee for display purposes
```

Comment:
"display purposes" might be a bit of a stretch, this is really used as view-layer logic, hence will block users from continuing if they don't have enough funds for the fast fees - which isn't a bad heuristic safety-wise, always good to plan a buffer.

I'd probably revamp that comment to say "view-layer logic purposes" or similar though. At first reading this, I thought of this as "display the fast fee, but use the average fee under the hood for everything else including has enough native token for gas calculations", but double-checking consumption proved otherwise

Reply:
https://github.com/shapeshift/web/pull/4324/commits/967fcf283682f83ae35dc3112cdbf363df1cbc31